src/nodes.ts

```diff
@@ -102,11 +106,23 @@ export class CosmosDBResourceNode implements IMongoServer {
 		return this._connectionString;
 	}
 
+	async getMasterKey(): Promise<string> {
+		const docDBClient = new DocumentdbManagementClient(this._subscriptionFilter.session.credentials, this._subscriptionFilter.subscription.subscriptionId);
+		const result = await docDBClient.databaseAccounts.listKeys(this._resourceGroupName, this._databaseAccount.name);
+		return result.primaryMasterKey;
+	}
+
 	async getChildren(): Promise<INode[]> {
 		if (this._isMongo) {
 			const connectionString = await this.getConnectionString();
 			return MongoServerNode.getMongoDatabaseNodes(connectionString, this);
 		}
+		if (this._isDocDB) {
+			const masterKey = await this.getMasterKey();
+			let client = new DocumentDBConnectionClient(this._databaseAccount.documentEndpoint, {masterKey: masterKey});
+			let DocDBServer = new DocDBServerNode(<string>masterKey, client.id, this._databaseAccount.documentEndpoint);
```

Comment:
There's no need to create a node just to call getChildren. I would prefer if we called a _static_ method on DocDBServerNode and just pass in the masterKey and endpoint

(Right now there's no need to _ever_ call the constructor for DocDBServerNode. The constructor for the MongoServerNode is only called in the 'Attach' scenario, but we don't support that yet for DocDB servers)